src/Compute/v2/Models/Keypair.php

```diff
@@ -0,0 +1,79 @@
+<?php declare (strict_types = 1);
+
+namespace OpenStack\Compute\v2\Models;
+
+use OpenCloud\Common\Resource\Creatable;
+use OpenCloud\Common\Resource\OperatorResource;
+use OpenCloud\Common\Resource\Deletable;
+use OpenCloud\Common\Resource\Listable;
+use OpenCloud\Common\Resource\Retrievable;
+use OpenCloud\Common\Transport\Utils;
+
+/**
+ * Represents a Compute v2 Keypair
+ *
+ * @property \OpenStack\Compute\v2\Api $api
+ */
+class Keypair extends OperatorResource implements Listable, Retrievable, Deletable, Creatable
+{
+    /** @var string */
+    public $fingerprint;
+
+    /** @var string */
+    public $name;
+
+    /** @var string */
+    public $publicKey;
+
+    /** @var  boolean */
+    public $deleted;
+
+    /** @var  string */
+    public $userId;
+
+    /** @var  string */
+    public $id;
+
+    /** @var \DateTimeImmutable */
+    public $createdAt;
```

Comment:
`updatedAt` and `deletedAt` are nullable thus this will throw exception https://github.com/php-opencloud/common/blob/master/src/Common/Resource/AbstractResource.php#L92

Perhaps `AbstractResource` should check if `$val` is null?